packages/connect-form/src/components/advanced-options-tabs/advanced-tab/advanced-tab.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { ChangeEvent, useCallback } from 'react';
+import { css } from '@emotion/css';
+import { ConnectionOptions } from 'mongodb-data-service';
+import {
+  RadioBox,
+  RadioBoxGroup,
+  spacing,
+  TextInput,
+  Label,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+import { MongoClientOptions } from 'mongodb';
+
+import FormFieldContainer from '../../form-field-container';
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import { ConnectionFormError } from '../../../utils/connect-form-errors';
+import { readPreferences } from '../../../utils/read-preferences';
+
+import UrlOptions from './url-options';
+
+const containerStyles = css({
+  marginTop: spacing[3],
+});
+
+const fieldStyles = css({
+  width: '50%',
+});
+
+function AdvancedTab({
+  updateConnectionFormField,
+  connectionStringUrl,
+}: {
+  errors: ConnectionFormError[];
+  connectionStringUrl: ConnectionStringUrl;
+  hideError: (errorIndex: number) => void;
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions?: ConnectionOptions;
+}): React.ReactElement {
+  const handleFieldChanged = useCallback(
+    (key: keyof MongoClientOptions, value: unknown) => {
+      if (!value) {
+        return updateConnectionFormField({
+          type: 'delete-search-param',
+          key,
+        });
+      }
+      return updateConnectionFormField({
+        type: 'update-search-param',
+        currentKey: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  const readPreference = connectionStringUrl.searchParams.get('readPreference');
```

Comment:
I noticed that while this setting auto updates when the connection string is being edited and this tab is just opened, after it's changed with the button and then the connection string is edited it no longer updates. Is this a case we can easily cover? Here's an example:

https://user-images.githubusercontent.com/1791149/148422244-6bcb0c38-f1d1-4573-84b8-893c491e6da6.mp4

Reply:
awesome catch 🔥